mapflow/functional/layer_utils.py

```diff
@@ -529,10 +530,36 @@ def download_results_callback(self, response: QNetworkReply, processing) -> None
             )
         if error:
             self.message_bar.pushWarning(self.tr("Error"),
-                                         self.tr('Failed to save results to file. '
-                                                 'Error code: {code}. Message: {message}').format(code=error,
-                                                                                                  message=msg))
-            return
+                                self.tr('Failed to save results to GeoPackage. '
+                                        'Error code: {code}. Message: {message}. ' 
+                                        'File will be saved as GeoJSON instead.').format(code=error, message=msg))
+            # Save as GeoGSON instead of GeoPackage
+            output_path = os.path.join(self.dlg.outputDirectory.text(), processing.id_)
+            extension = '.geojson'
+            if os.path.exists(output_path + extension):
+                count = 1
+                while os.path.exists(output_path + f'({count})' + extension):
+                    count += 1
+                output_path += f'({count})' + extension
+            else:
+                output_path += extension
+            write_options.driverName = "GeoJSON"
+            with open(os.path.join(self.temp_dir, os.urandom(32).hex()), mode='wb+') as f:
+                f.write(response_data)
+                f.seek(0)
+                layer = QgsVectorLayer(f.name, '', 'ogr')
+                error_geojson, msg_geojson, *_ = QgsVectorFileWriter.writeAsVectorFormatV3(
```

Comment:
We don't need to write the data here using qgis driver, as the response is valid geojson

Use this sample:
https://github.com/Geoalert/mapflow-qgis/blob/2f16fa0b177baf7ee9251e47c1e00d980492ec10/mapflow/functional/layer_utils.py#L447